diff --git a/ext/standard/php_fopen_wrapper.c b/ext/standard/php_fopen_wrapper.c
index 6d7e77d1b5..f4b3a6de99 100644
--- a/ext/standard/php_fopen_wrapper.c
+++ b/ext/standard/php_fopen_wrapper.c
@@ -313,5 +313,5 @@ php_stream *php_stream_url_wrap_php(php_stream_wrapper *wrapper, const char *pat
-#ifdef HAVE_UNISTD_H
+#if defined(HAVE_UNISTD_H) && defined(HAVE_GETDTABLESIZE)
 		dtablesize = getdtablesize();
 #else
 		dtablesize = INT_MAX;
 #endif
diff --git a/ext/standard/proc_open.c b/ext/standard/proc_open.c
index 4b5ac1cf9a..36e3b2d0b5 100644
--- a/ext/standard/proc_open.c
+++ b/ext/standard/proc_open.c
@@ -1396,7 +1396,7 @@ PHP_FUNCTION(proc_open)
 	childHandle = pi.hProcess;
 	child       = pi.dwProcessId;
 	CloseHandle(pi.hThread);
-#elif defined(USE_POSIX_SPAWN)
+#elif defined(USE_POSIX_SPAWN) && !defined(__EMSCRIPTEN__)
 	posix_spawn_file_actions_t factions;
 	int r;
 	posix_spawn_file_actions_init(&factions);
